packages/backend/src/controllers/membership.ts

```diff
@@ -1,43 +1,190 @@
+   * or all if the user is admin
+   * @param req
+   * @param res
+   */
+  public getAll = async (req: Request, res: Response): Promise<Response> => {
+    if (req.user.user.group === EXISTING_GROUPS.ADMIN) {
+      return res.send(await this.findAll());
+    }
+
+    const currentOrg = await ControllerUtils.getCurrentOrgFromUserInRequest(
+      req
+    );
+
+    const requestResult = await this.repository
+      .createQueryBuilder('membership')
+      .innerJoin('membership.member', 'member')
+      .where('member.organisationId = :orgId', { orgId: currentOrg.id })
+      .getMany();
+
+    return res.send(requestResult);
+  };
+
+  /**
+   * Custom getOne that return the membership with the id that belongs to
+   * the user organisationor all if the user is admin
+   * @param req
+   * @param res
+   */
+  public getOne = async (req: Request, res: Response): Promise<Response> => {
+    const id = Number.parseInt(req.params.id);
+
+    const membership = await this.repository.findOne(id, {
+      relations: ['paymentRequest', 'paymentRequest.payment', 'member']
     });
-    const withoutPayment = memberships.filter(
-      (membership) => !membership.paymentRequest?.payment
+
+    if (!membership) {
+      throw new APIError(404, 'not found');
+    }
+
+    const currentOrg = await ControllerUtils.getCurrentOrgFromUserInRequest(
+      req
     );
+
+    if (
+      req.user.user.group !== EXISTING_GROUPS.ADMIN &&
+      membership.member.organisation.id !== currentOrg.id
+    ) {
+      throw new APIError(403, 'unauthorized');
+    }
+
+    return res.send(membership);
+  };
+
+  /**
+   * Will get all the membership that are not paid
+   * @param req
+   * @param res
+   */
+  public getNotPaid = async (
+    req: Request,
+    res: Response
+  ): Promise<Response> => {
+    const today: Date = new Date();
+
+    const currentOrg = await ControllerUtils.getCurrentOrgFromUserInRequest(
+      req
+    );
+
+    const membershipRequest = this.repository
+      .createQueryBuilder('membership')
+      .innerJoinAndSelect('membership.member', 'member')
+      .innerJoinAndSelect('member.user', 'memberUser')
+      .leftJoinAndSelect('membership.paymentRequest', 'paymentRequest')
+      .leftJoinAndSelect('paymentRequest.payment', 'paymentRequestPayment')
+      .innerJoinAndSelect('membership.plan', 'plan')
+      .where('membership.endDate <= :today', {
+        today: today.toDateString()
+      });
+
+    if (req.user.user.group !== EXISTING_GROUPS.ADMIN) {
+      membershipRequest.andWhere('member.organisationId = :orgId', {
+        orgId: currentOrg.id
+      });
+    }
+
+    const memberships = await membershipRequest.getMany();
+
+    const withoutPayment = memberships
+      // will get only not paid payment
+      .filter((membership) => !membership.paymentRequest?.payment)
+      // will remove all password from user
+      .map((membership) => {
+        delete membership.member.user.password;
```

Comment:
It would be better to change user'entity and ad an option to never retrieve passord ?

(I know it's not your code, muahah)

Reply:
The better solution would be to not select the password field. But for now, it is not possible to do that in TypeORM (https://github.com/typeorm/typeorm/issues/535) as we can only choose which field we want to select (and not omit)

Reply:
Resolved in last commit